x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/ExactMatchProcessor.java

```diff
@@ -30,62 +28,66 @@
                         Client client,
                         String policyName,
-                        String enrichKey,
+                        String field,
+                        String targetField,
+                        String matchField,
                         boolean ignoreMissing,
-                        boolean overrideEnabled,
-                        List<EnrichSpecification> specifications) {
+                        boolean overrideEnabled) {
         this(
             tag,
             createSearchRunner(client),
             policyName,
-            enrichKey,
-            ignoreMissing,
-            overrideEnabled,
-            specifications
+            field,
+            targetField,
+            matchField, ignoreMissing,
+            overrideEnabled
         );
     }
 
     ExactMatchProcessor(String tag,
                         BiConsumer<SearchRequest, BiConsumer<SearchResponse, Exception>> searchRunner,
                         String policyName,
-                        String enrichKey,
+                        String field,
+                        String targetField,
+                        String matchField,
                         boolean ignoreMissing,
-                        boolean overrideEnabled,
-                        List<EnrichSpecification> specifications) {
+                        boolean overrideEnabled) {
         super(tag);
         this.searchRunner = searchRunner;
         this.policyName = policyName;
-        this.enrichKey = enrichKey;
+        this.field = field;
+        this.targetField = targetField;
+        this.matchField = matchField;
         this.ignoreMissing = ignoreMissing;
         this.overrideEnabled = overrideEnabled;
-        this.specifications = specifications;
     }
 
     @Override
     public void execute(IngestDocument ingestDocument, BiConsumer<IngestDocument, Exception> handler) {
         try {
             // If a document does not have the enrich key, return the unchanged document
-            final String value = ingestDocument.getFieldValue(enrichKey, String.class, ignoreMissing);
+            final String value = ingestDocument.getFieldValue(field, String.class, ignoreMissing);
             if (value == null) {
                 handler.accept(ingestDocument, null);
                 return;
             }
 
-            TermQueryBuilder termQuery = new TermQueryBuilder(enrichKey, value);
+            TermQueryBuilder termQuery = new TermQueryBuilder(field, value);
             ConstantScoreQueryBuilder constantScore = new ConstantScoreQueryBuilder(termQuery);
             SearchSourceBuilder searchBuilder = new SearchSourceBuilder();
             searchBuilder.from(0);
             searchBuilder.size(1);
             searchBuilder.trackScores(false);
-            searchBuilder.fetchSource(specifications.stream().map(s -> s.sourceField).toArray(String[]::new), null);
+            searchBuilder.fetchSource(null, matchField);
```

Comment:
I see, lets then not filter out the match field. If filtering out the match field is required then this can be done by another processor configured after the enrich processor.